src/hooks/sync-sites/use-sync-push.ts

```diff
@@ -129,9 +129,13 @@ export function useSyncPush( {
 				updatePushState( selectedSite.id, remoteSiteId, {
 					status: pushStatesProgressInfo.failed,
 				} );
+				let errorMessage = __( 'Studio was unable to connect to WordPress.com. Please try again.' );
+				if ( typeof error === 'object' && error !== null && 'error' in error ) {
+					errorMessage = ( error as { error: string } ).error || errorMessage;
+				}
```

Comment:
```suggestion
				if ( typeof error?.error === 'string' && error.error ) {
					errorMessage = error.error;
				}
```

I did some digging to try and find a better way to identify if `error` is of the right type. I couldn't find anything great, but I still think we can improve the current solution slightly.

Reply:
Thank you for your suggestion, @fredrikekelund. Originally, I had a simpler check in place, but TS was not happy about it. Even though I prefer the suggested change, it still triggers the following error:

```
Property 'error' does not exist on type '{}'.
```

Every alternative I have tried so far is quite verbose too. 🤔

Reply:
Would it make sense to move this logic to a separate function so that instead of redefining the variables, we could rely on early returns?

Reply:
> TS was not happy about it

Seems like I overlooked the TS errors in my suggestion yesterday… 

> move this logic to a separate function

Yeah, adding a `getErrorMessageFromResponse` function, or alternatively, an `isDecoratedErrorResponse` function seems like a good idea. It would help clean up this code. Those function names are obviously just suggestions.

Reply:
Thank you Wojtek and Fredrik. I have extracted out the logic in the latest commit https://github.com/Automattic/studio/pull/711/commits/9439adeeeab4fb6ae879b2a89cce602965a2f4cf. Feel free to let me know what you think.

Reply:
Just to note that I have now simplified the logic even further based on Wojtek's suggestion.